script.js

```diff
@@ -1,4 +1,285 @@
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    // Store the current suit in a variable
+    var currentSuit = suits[suitIndex];
+
+    // Loop from 1 to 13 to create all cards for a given suit
+    // Notice rankCounter starts at 1 and not 0, and ends at 13 and not 12.
+    // This is an example of a loop without an array.
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      // By default, the card name is the same as rankCounter
+      var cardName = rankCounter;
+
+      // If rank is 1, 11, 12, or 13, set cardName to the ace or face card's name
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+      } else if (cardName == 12) {
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+
+      // Create a new card with the current name, suit, and rank
+      var card = {
+        cardName: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+
+      // amend the card rank for jack queen king to be 10 so that score of these cards will all be 10
+      if (card.rank == 11) {
+        card.rank = 10;
+      } else if (card.rank == 12) {
+        card.rank = 10;
+      } else if (card.rank == 13) {
+        card.rank = 10;
+      }
+
+      // Add the new card to the deck
+      cardDeck.push(card);
+
+      // Increment rankCounter to iterate over the next rank
+      rankCounter += 1;
+    }
+
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+
+  // Return the completed card deck
+  return cardDeck;
+};
+// define gamemodes in blackjack
+var DEAL_CARDS = `deal`;
+var HIT = `hit`;
+var STAND = `stand`;
+var DEALER_TURN = `dealer's turn`;
+
+// gameMode at initial stage
+var gameMode = DEAL_CARDS;
+
+// shuffle deck of cards
+var deck = makeDeck();
+var shuffledDeck = shuffleCards(deck);
+
+var playerCurrScore;
+var dealerCurrScore;
+var playerHand = [];
+var displayPlayerCardArray = [];
+var dealerHand = [];
+var displayDealerCardArray = [];
+var maxScore = 21;
+var dealerMinimalScore = 17;
+var cardName;
+var rank;
+
+// standard messages
+var hitOrStandMessage =
+  "Enter hit to take another card or stand to end your turn.";
+var refreshGameMessage = " Refresh to start new game";
+
+// display cards in hand to show player the cards drawn so far
+// reference https://stackoverflow.com/questions/19590865/from-an-array-of-objects-extract-value-of-a-property-as-array
+var displayCardsInHand = function (handArray, displayCardsArray, cardName) {
+  displayCardsArray = [];
+  for (var counter = 0; counter < handArray.length; counter += 1) {
+    displayCardsArray.push(handArray[counter].cardName);
+  }
+  return displayCardsArray;
+};
+
+// https://stackoverflow.com/questions/1230233/how-to-find-the-sum-of-an-array-of-numbers
+
+// +++cal total score first then filter out ace value+++
+var calTotalScore = function (array, rank) {
```

Comment:
array can be better name as cards.
try to avoid javascript types as the variable names especially when they can be better described
the rank input doesn't seem used, can be removed